src/include/miopen/graphapi/engine.hpp

```diff
@@ -49,8 +51,52 @@ class Engine
     Engine(const Solution& solution) : mSolution(solution) {}
     Engine(Solution&& solution) : mSolution(std::move(solution)) {}
 
-    const Solution& getSolution() const { return mSolution; }
-    Solution& getSolution() { return mSolution; }
+    const Solution& getSolution() const noexcept { return mSolution; }
+    Solution& getSolution() noexcept { return mSolution; }
+
+    int64_t getGlobalIndex() const noexcept { return mGlobalIndex; }
+    int32_t getSmCount() const noexcept { return mSmCount; }
+};
+
+class OpGraph;
+
+class EngineBuilder
+{
+private:
+    const OpGraph* mOpGraph = nullptr;
+    int64_t mGlobalIndex    = -1;
+    int32_t mSmCount        = 0;
+    bool mGlobalIndexSet    = false;
+
+public:
+    EngineBuilder& setOpGraph(const OpGraph* opGraph);
```

Comment:
engine needs to keep a graph pointer. I can add it in my latest PR.

Reply:
If your PR depends on it, add it in your PR. Also I can add it in my next [GTEST] PR.